### ISSUE TITLE
Selection scan power?

### ISSUE THREAD

**Issue by mpinsky** [Rosemeis/pcangsd#39]
Hi all,

I'm curious if anyone has run a power analysis on the pcangsd selection scan or has thoughts on the below graph. Also let me know if there's a better place to post this query (a discussion group?). 

I've done a few simulations with SLiM of a 30Mb chromosome in which one SNP in the middle has positive selection applied (s from 0 to 1.5, the x-axis below). Population sizes are from 100 to 10,000 diploids (columns in the graph below), and the selected SNP is at an initial frequency of 0.05 or 0.2 (colors). The simulations run for 11 generations and the pcangsd selection scan is run with samples from generation 1 and from generation 11. Pcangsd is run on either just one chromosome (top row of the graph below) or a full genome in which the chromosome with selection has been combined with 20 extra chromosomes without any selection (bottom row). 

What is odd is that there appears to be little to no power for larger population sizes (3000 or 10,000), where drift is quite weak and selection quite strong and otherwise easy to detect. 

The y-axis here is the minimum -log10 p-value for any SNP from the pcangsd selection scan.

![slim_pcangsdoutlier_minp](https://user-images.githubusercontent.com/6370532/84387455-4dddb700-abf3-11ea-91e8-8684dbb82caf.png)

**Comment by Rosemeis** [comment]
Hi,

The selection statistic is based on the one from Galinsky et al. 2016 (https://www.cell.com/ajhg/fulltext/S0002-9297(16)00003-3), where we use posterior dosages instead of genotypes for low coverage data. They also have some power analyses. :-)

So it is used to detect selection between populations with continuous structure by testing differentiation of each site along PCs capturing population structure. I assume that you are only simulating one population here but I might be wrong.

Best,
Jonas

**Comment by mpinsky** [comment]
Hey Jonas, thanks. The Galinsky power analysis is definitely useful, though I'm looking at smaller sample sizes (non-model system, about 25 individuals per sample). 

In the graph I posted above, you're correct, I'm only simulating one population. The two samples are separated in time (11 generations) with selection that starts at the same time the first sample is taken. 

From FST plots, it's clear there is some genome-wide divergence over those 11 generations, but not much. In contrast, the SNP under selection (in the middle of the chromosome) changes frequency dramatically and pulls some of the nearby linked SNPs with it. Here's an example for Ne=10000 and s=1.5, where pcangsd appears not to detect any selection:

![image](https://user-images.githubusercontent.com/6370532/84628639-53841700-aee9-11ea-96a0-88f2cf958329.png)

**Comment by Rosemeis** [comment]
It is definitely an interesting scenario that I haven't thought about (in regards to a temporal component)! :-)

How does your PCA look like? For it to be optimal I would only use 1 eigenvector in the PCAngsd (if it doesn't already do that automatically) "-e 1". So you would need the first PC axis to separate the two samples.

**Comment by mpinsky** [comment]
For all the whole genome simulations, PCAngsd is only choosing one eigenvector anyway. Here's an example for Ne=10000, showing very little or no divergence, as you'd expect:
![image](https://user-images.githubusercontent.com/6370532/84655107-a3c29f80-af10-11ea-9cdc-1a7a9eb5ac1b.png)

But at Ne=100, you get quite a bit, as expected:
![image](https://user-images.githubusercontent.com/6370532/84655084-9d342800-af10-11ea-8749-2295655f2114.png)

I've only been assessing the selection test statistic along the first PC.

**Comment by Rosemeis** [comment]
Okay in that case, it does make sense that PCAngsd is struggeling with simulations of higher effective population sizes, as it can't distinguish the two samples. The selection statistic is meant to test for differentiation along the PC axis, so if it doesn't separate the samples then it doesn't have the power to detect it.

**Comment by mpinsky** [comment]
Yes, that must be it, and great point that is probably obvious to those developing the methods, but perhaps not as immediately clear when picking up a new method. PC1 may explain the "most" variation across samples, but when sampling noise is high (e.g., small sample sizes) and differentiation is weak (e.g., large Ne), that axis becomes noise as well. And so any outliers along the axis are meaningless if they exist at all. 

This is when an FST-based test works better because one can clearly define the two samples.

**Comment by Rosemeis** [comment]
That is true. Thank you for looking a bit into this though! :-)

**Comment by mpinsky** [comment]
Hi Jonas,

I'm re-opening this to see if some new results make sense. I've pushed the previous power analyses to larger Ne values and have seen a transition from no power (as discussed previously in this thread) at moderately high Ne (3,000 to 10,000) to many false positives with a full genome at Ne = 30000:

![slim_pcangsdoutlier_minp](https://user-images.githubusercontent.com/6370532/99245925-6f5e5280-27d2-11eb-9928-18110a6f9483.png)

This is the same plot and same simulation setup as in the first plot in this thread, but as a refresher: y-axis is the minimum -log10(p value) from PCAngsd, the x-axis is the selection coefficient, the columns are Ne (labeled at the top), and the rows are either a single 30 Mb chromosome (top row) or a full genome (bottom row, 21 chromosomes). The colors are the initial frequency of the SNP under selection. After a burn-in, a sample of the single population is taken, then the simulation in SLiM runs with selection for eleven generations, and finally a second sample is taken. PCAngsd is run on a comparison of the two samples through time.

For the false positives, I'm looking in particular at Ne=30000, s=0, full genomes (bottom row). All the simulations for those parameters have low p-values. 

Plotting the proportion of simulations with FDF-adjusted p-values < 0.05 shows it clearly:

![slim_pcangsdoutlier_minp](https://user-images.githubusercontent.com/6370532/99246923-18f21380-27d4-11eb-9396-9b003bc4b511.png)

Since the issue shows up for full genomes, my best guess for the switch in behavior is that we have a lot of SNPs at high Ne and therefore regain an ability to differentiate the samples:

![slim_nsnps](https://user-images.githubusercontent.com/6370532/99246466-56a26c80-27d3-11eb-8abb-69a307b512d1.png)

Right column is full genomes. 

A sample of PCA plots shows some population differentiation at Ne=300 and 1000, very little at 3000 and 10000, but then some again at 30000:

![slim_pca_n300_s0_f0 05_i1_wholegen](https://user-images.githubusercontent.com/6370532/99248088-eea15580-27d5-11eb-90a4-78bd850675c9.png)
![slim_pca_n1000_s0_f0 05_i1_wholegen](https://user-images.githubusercontent.com/6370532/99248090-eea15580-27d5-11eb-8738-6ae5788ebdec.png)
![slim_pca_n3000_s0_f0 05_i1_wholegen](https://user-images.githubusercontent.com/6370532/99248091-eea15580-27d5-11eb-9b3e-b07caa7fcd20.png)
![slim_pca_n10000_s0_f0 05_i1_wholegen](https://user-images.githubusercontent.com/6370532/99248092-ef39ec00-27d5-11eb-8908-a879920c4f8e.png)
![slim_pca_n30000_s0_f0 05_i1_wholegen](https://user-images.githubusercontent.com/6370532/99248095-ef39ec00-27d5-11eb-833d-2915def69a64.png)

If you have any thoughts, I'd be very curious!
